### PR TITLE
Add newline to the end of generated files

### PR DESCRIPTION
**Pull request by jhrcek**
Followup on the huge PR I just saw merged: https://github.com/haskell/lsp/pull/478 ( CC @michaelpj)
Let's not forget newline at the end of generated files.

- Many Unix/Linux tools expect or require this newline at the end of files. If it's missing, these tools might not be able to process the last line correctly, or they might throw warnings or errors.
- The POSIX standard, which defines the core of Unix/Linux operating systems, requires each line to end with a newline character.

### CODE REVIEW

File: lsp-types/generator/CodeGen.hs [hunk 1]
```diff
@@ -106,7 +106,13 @@ genModule name pragmas mexports action = do
...
       isSelfImport imp = (" " <> fullModName <> " ") `T.isInfixOf` imp || (" " <> fullModName) `T.isSuffixOf` imp
       importSection = hardvcat (fmap pretty $ filter (not . isSelfImport) $ toList imports)
-      mod = warning <> hardline <> pragmaSection <> hardline <> optionsSection <> hardline <> header <> hardline <> hardline <> importSection <> hardline <> hardline <> doc
+      mod = warning <> hardline
+          <> pragmaSection <> hardline
+          <> optionsSection <> hardline
+          <> header <> hardline <> hardline
+          <> importSection 
+          <> hardline <> hardline 
+          <> doc <> hardline
```
Comment (jhrcek) [inline lsp-types/generator/CodeGen.hs:115]: The actual change is all the way down here: addition of hardline at the end of generated files.

Comment (michaelpj) [reply lsp-types/generator/CodeGen.hs:115]: Thank you!

**Review by michaelpj — APPROVED** [review]
Thanks for catching this!

**Comment by jhrcek** [comment]
My pleasure. Thank you for implementing the generator logic. Hopefully it will make updating to newer LSP versions easier!

**Pull request merged**